integration_tests/src/main/python/conditionals_test.py

```diff
@@ -296,3 +296,85 @@ def test_conditional_with_side_effects_unary_minus(data_gen, ansi_enabled):
             'CASE WHEN a > -32768 THEN -a ELSE null END'),
         conf = {'spark.sql.ansi.enabled': ansi_enabled})
 
+_case_when_scalars = [
```

Comment:
Is it possible to leverage the `data_gen` to build a more robust test case just as other test?

Reply:
In this case `test_case_when_all_then_values_are_scalars`, for the bool columns, the `data_gen` is as following:
```
    data_gen = [
        ("a", boolean_gen),
        ("b", boolean_gen),
        ("c", boolean_gen),
        ("d", boolean_gen),
        ("e", boolean_gen)
    ]

    sql =  """
            select case
                when a then {}
                when b then {}
                when c then {}
                when d then {}
                else {}
            end
            from tab
            """
```
The `_case_when_scalars` specify the const scalars, `data_gen` can not be used.